R/FilterStates.R

```diff
@@ -385,7 +385,8 @@ FilterStates <- R6::R6Class( # nolint
                 observeEvent(
                   eventExpr = fs_callback(), # when remove button is clicked in the FilterState ui
                   once = TRUE, # remove button can be called once, should be destroyed afterwards
-                  handlerExpr = private$state_list_remove(state$get_state()$id)
+                  handlerExpr = private$state_list_remove(state$get_state()$id),
+                  ignoreInit = TRUE
```

Comment:
**This is a protection against future implementations of new `FilterStateXXXX` that don't reset the inputs.**

This line alone fixes the problem, as we don't want to perform any removal action as soon as the `$server()` is called.

What happened before is that there was no NULLification of all the inputs _(via `destroy_shiny()`)_ and when the UI for the slice was created again, it had value `1`, hence `private$state_list_remove(state$get_state()$id)` was called.

So, as soon as it was added it would be removed.

Reply:
Understood :+1: